app/src/main/java/com/example/designpatternstudy/TemplateMethod/FoodMaker.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.designpatternstudy.TemplateMethod
+
+enum class DelieverDistance {
+    Short,
+    Long
+}
+
+abstract class FoodMaker {
+
+    fun prepareFood(distance: DelieverDistance) {
+        noticePrepare()
+        prepareCondiments()
+        cutCondiments()
+        cook()
+
+        when (distance) {
+            DelieverDistance.Short -> {
+                boxingLowDistance()
+            }
+            DelieverDistance.Long -> {
+                boxingLongDistance()
+            }
+        }
+
+        if(IsChicken()){
+            putSideCooking()
+        }
+
+    }
+
+    private fun noticePrepare() {
+        System.out.println("음식을 준비하겠습니다.")
+    }
+
+    protected abstract fun prepareCondiments()
+    protected abstract fun cook()
+    protected abstract fun cutCondiments()
+    protected abstract fun putSideCooking()
```

Comment:
```suggestion
    protected open fun putSideCooking(){}
```
Template method pattern의 경우 template method, 여기선 `prepareFood`와 step으로 구성됩니다.  step의 경우 기본 구현을 제공하고 필수 구현인 경우에만 abstract로 작성하는데 FoodMaker에서는 private이나 abstract로 작성하신것으로 보여지네요. 
`IsChicken`은 없애고 putSideCooking을 open 으로 작성하는것이 좀 더 적절해보입니다.